packages/eui/src/components/date_picker/react-datepicker/src/calendar.js

```diff
@@ -352,15 +363,35 @@ export default class Calendar extends React.Component {
         </div>
       );
     }
+
+    const isLocaleEnglish = this.props.locale
+      ? Array.isArray(this.props.locale.match(/^en\b/))
+      // when this.props.locale is not set the default is 'en' 
+      : this.props.locale === undefined;
```

Comment:
See above, my 2c for simplifying this
```suggestion
    const isLocaleEnglish = !this.props.locale || this.props.locale.startsWith('en');
```